pkg/goDB/encoder/lz4/lz4_native.go

```diff
@@ -0,0 +1,70 @@
+//go:build !cgo || goprobe_noliblz4
+// +build !cgo goprobe_noliblz4
+
+package lz4
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+// Close will close the encoder and release potentially allocated resources
+func (e *Encoder) Close() error {
+	return nil
+}
+
+// Compress compresses the input data and writes it to dst
+func (e *Encoder) Compress(data, buf []byte, dst io.Writer) (n int, err error) {
+
+	// Handle output slice size
+	dstCapacity := lz4.CompressBlockBound(len(data))
+	if cap(buf) < dstCapacity {
+		buf = make([]byte, 0, 2*dstCapacity)
+	}
+	buf = buf[:dstCapacity]
+
+	// Compress data
+	compLen, err := lz4.CompressBlockHC(data, buf, lz4.CompressionLevel(e.level), nil, nil)
+	if err != nil {
+		return n, fmt.Errorf("lz4: compression failed: %s (%d)", err, compLen)
```

Comment:
Just to poke you a bit. There's too many direct mentions of `lz4:` in these messages ;). I'd suggest two options:

1. the quick one: prefix with an `lz4` const and provide convenience method for error
2. an `lz4Error` type implementing `Error`, appending length data if available and doing the prefixing

```golang
type lz4Error struct {
  err error
  len int
}
```

Reply:
I've opted for a third approach:

- Drop `lz4` / `zstd` prefix altogether (after all, we're not doing that anywhere else, why here - it should be clear from the calling context / logging / error handling which compression algorithm is used, so an additional string prefix for each error is not / should not be necessary)
- Refactor fixed errors as variable declarations `var ErrSomethingSomething = ...` and reuse them